File: implementations/php/src/sqlite_serial.php
<?php

include "gaussian.php";

try {
  $dbPath = sprintf("sqlite:%s/../../sqlite/database.db", dirname(__FILE__));
  $conn = new PDO($dbPath);
  $conn->exec("DROP TABLE IF EXISTS tb_php_serial; CREATE TABLE tb_php_serial(id INTEGER PRIMARY KEY 
  AUTOINCREMENT, z_score REAL NOT NULL, cumulative_distribution REAL NOT NULL)");
  for ($i = -500; $i <= 500; $i++) {
    $x = $i / 100;
    $prob = gaussianCDF(0, 1, $x);
    $query = sprintf("INSERT INTO tb_php_serial(z_score, cumulative_distribution) VALUES (%.2f, %f);", $x, $prob);
    $conn->exec($query);
  }
} catch (\Throwable $th) {
  echo $th->getMessage();
}

File: implementations/c/src/sqlite-serial.c
#include <stdio.h>
#include <libgen.h>
#include <stdlib.h>
#include <stdint.h>
#include <sqlite3.h>
#include <hash.h>

#define N 10000

int main(int argc, char **argv)
{
  sqlite3 *conn = NULL;
  int32_t i, hash;
  int exitCode = -1;
  char query[90],
      *dbPath = "database.db",
      *errMsg = NULL;

  exitCode = sqlite3_open(dbPath, &conn);
  if (exitCode)
  {
    fprintf(stderr, "Connection to database failed\n");
    goto cleanup;
  }

  exitCode = sqlite3_exec(conn, "DROP TABLE IF EXISTS tb_c_serial; CREATE TABLE tb_c_serial "
                                "(id INTEGER PRIMARY KEY AUTOINCREMENT, hash INTEGER NOT NULL)",
                          0, 0, &errMsg);
  if (exitCode)
  {
    fprintf(stderr, "%s\n", errMsg);
    goto cleanup;
  }
  sqlite3_free(errMsg);
  errMsg = NULL;

  for (i = 1; i <= N; i++)
  {
    hash = hash32(i);
    snprintf(query, 90, "INSERT INTO tb_c_serial(hash) VALUES (%d);", hash);

    exitCode = sqlite3_exec(conn, query, 0, 0, &errMsg);
    if (exitCode)
    {
      fprintf(stderr, "%s\n", errMsg);
      goto cleanup;
    }
    sqlite3_free(errMsg);
    errMsg = NULL;
  }

  exitCode = 0;

cleanup:
  sqlite3_free(errMsg);
  if (conn)
    sqlite3_close(conn);

  return exitCode;
}
